src/robot/parsing/populators.py

```diff
@@ -199,17 +199,27 @@ def _extension_is_accepted(self, ext, incl_extensions):
             return ext in incl_extensions
         return ext in READERS
 
-    def _is_included(self, path, base, ext, incl_extensions, incl_suites):
+    def _is_included(self, path, datadir, base, ext, incl_extensions, incl_suites):
         if base.startswith(self.ignored_prefixes):
             return False
         if os.path.isdir(path):
             return base not in self.ignored_dirs or ext
         if not self._extension_is_accepted(ext, incl_extensions):
             return False
-        return self._is_in_included_suites(base, incl_suites)
+        return self._is_in_included_suites(base, incl_suites, self._get_longname(base, datadir, incl_suites))
 
-    def _is_in_included_suites(self, name, incl_suites):
+    def _is_in_included_suites(self, name, incl_suites, longname=None):
+        if longname:
+            return not incl_suites or incl_suites._match_longname(longname)
         return not incl_suites or incl_suites.match(self._split_prefix(name))
 
     def _split_prefix(self, name):
         return name.split('__', 1)[-1]
+
+    def _get_longname(self, base, datadir, include_suites):
+        if datadir.parent and len(include_suites._matcher) > 1:
```

Comment:
Accessing private attribute `include_suites._matcher` is not OK. Also, I don't understand why it must have more than one entry. If more than zero entries were fine, it would be possible to use just `if datadir.parent and include_suites:`.